hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDiffUtils.java

```diff
@@ -73,44 +71,38 @@ public static String constructBucketKey(String keyName) {
   }
 
   public static void filterRelevantSstFiles(Set<String> inputFiles,
-      Map<String, String> tableToPrefixMap) throws IOException {
-    for (Iterator<String> fileIterator =
-         inputFiles.iterator(); fileIterator.hasNext();) {
-      String filepath = fileIterator.next();
-      if (!RocksDiffUtils.doesSstFileContainKeyRange(filepath,
-          tableToPrefixMap)) {
-        fileIterator.remove();
-      }
-    }
+                                            Map<String, String> tableToPrefixMap,
+                                            ManagedRocksDB... dbs) {
+    filterRelevantSstFiles(inputFiles, tableToPrefixMap, Collections.emptyMap(), dbs);
   }
 
-  public static boolean doesSstFileContainKeyRange(String filepath,
-      Map<String, String> tableToPrefixMap) throws IOException {
-
-    try (
-        ManagedOptions options = new ManagedOptions();
-        ManagedSstFileReader sstFileReader = new ManagedSstFileReader(options)) {
-      sstFileReader.open(filepath);
-      TableProperties properties = sstFileReader.getTableProperties();
-      String tableName = new String(properties.getColumnFamilyName(), UTF_8);
-      if (tableToPrefixMap.containsKey(tableName)) {
-        String prefix = tableToPrefixMap.get(tableName);
-
-        try (
-            ManagedReadOptions readOptions = new ManagedReadOptions();
-            ManagedSstFileReaderIterator iterator = ManagedSstFileReaderIterator.managed(
-                sstFileReader.newIterator(readOptions))) {
-          iterator.get().seek(prefix.getBytes(UTF_8));
-          String seekResultKey = new String(iterator.get().key(), UTF_8);
-          return seekResultKey.startsWith(prefix);
+  /**
+   * Filter sst files based on prefixes.
+   */
+  public static void filterRelevantSstFiles(Set<String> inputFiles,
+                                            Map<String, String> tableToPrefixMap,
+                                            Map<String, CompactionNode> preExistingCompactionNodes,
+                                            ManagedRocksDB... dbs) {
+    Map<String, LiveFileMetaData> liveFileMetaDataMap = new HashMap<>();
+    int dbIdx = 0;
+    for (Iterator<String> fileIterator =
+         inputFiles.iterator(); fileIterator.hasNext();) {
+      String filename = FilenameUtils.getBaseName(fileIterator.next());
+      while (!preExistingCompactionNodes.containsKey(filename) && !liveFileMetaDataMap.containsKey(filename)
```

Comment:
1. I don't think we need nested while loop.
2. Can it be moved outside the for loop?

Reply:
We don't have to read all rocksdb. We can minimise number of rocksdb to be read.

Reply:
Once it gets inside the outer while loop, inside while loop will load liveFileMetaData from both snapshots.

Reply:
No it won't. It will only load the second snapshot if the file is not present in the first one or the compaction log

Reply:
Oh ok the second one was supposed to be an if

Reply:
done